### PR TITLE
✨ Nodes networking

### PR DESCRIPTION
**Pull request by BertrandD**
```
# Run a node that will create block from time to time
RUST_LOG=trace cargo run 

# Runs a 2nd node that connects to first one
RUST_LOG=trace cargo run -- --config-file slave.ron
```

### CODE REVIEW

File: src/ctx.rs [hunk 1]
```diff
@@ -67,16 +67,17 @@ impl Ctx {
...
 
-    pub async fn start(&mut self, mut rx: Receiver<CtxCommand>) {
+    pub async fn start(&mut self, mut rx: Receiver<CtxCommand>) -> anyhow::Result<()> {
         while let Some(msg) = rx.recv().await {
             match msg {
                 CtxCommand::AddTransaction(tx) => self.handle_tx(tx),
                 CtxCommand::GenerateNewBlock => self.new_block(),
                 CtxCommand::SaveOnDisk => {
-                    let _ = self.save_on_disk();
+                    self.save_on_disk()?;
```
Comment (hhalex) [inline src/ctx.rs:76]: an error here should stop message consumption ?

Comment (BertrandD) [reply src/ctx.rs]: right, merged too quickly

File: src/p2p/peer.rs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        self.stream
+            .write(msg.as_binary().as_ref())
+            .await
+            .context("Failed to write data on stream")?;
+        Ok(())
+    }
+
+    async fn read_next_message(&mut self, msg_size: u32) -> Result<(), Error> {
+        if msg_size == 0 {
+            return Err(anyhow!("Connection closed by remote (1)"));
```
Comment (hhalex) [inline src/p2p/peer.rs:94]: OPT: bail!("error msg") does this

File: src/ctx.rs (hunk 1 above)
Comment (BertrandD) [reply src/ctx.rs]: fixed in 05991b7

**Pull request merged**